### PR TITLE
Add alternative admin cli

### PR DESCRIPTION
**Pull request by lukpueh**
**Note to reviewer**
Apologies for the huge diff.  There's a lot of shared infrastructure for the 3 commands, so it made sense to PR it all at once. The implementation alone is 800 lines, the rest is static test data and very extensive unit tests.
```
git diff --stat main -- repository_service_tuf
 5 files changed, 796 insertions(+)
git diff --stat main -- tests/files
 11 files changed, 338 insertions(+)
git diff --stat main -- tests/unit/
 1 file changed, 533 insertions(+)
```

I'm happy to make an effort to split up the PR, if needed.

**Description of changes**

Added commands:

- `rstuf admin2 ceremony`
- `rstuf admin2 update`
- `rstuf admin2 sign`

Previous work, related discussion, and detailed reasons for the re-write can be found in https://github.com/repository-service-tuf/repository-service-tuf-cli/pull/477 and https://github.com/repository-service-tuf/repository-service-tuf-cli/pull/490. These PRs are superseded by this PR.

Currently, the cli does not interact with the RSTUF API, but reads input data from files passed as cli arguments (update, sign) and optionally writes output data to file using the `--payload-out` option (ceremony, update, sign).

Preliminary API integration can be found in https://github.com/repository-service-tuf/repository-service-tuf-cli/pull/477. I suggest to re-implement this in a separate PR, as well as any presentation improvements discussed in https://github.com/repository-service-tuf/repository-service-tuf-cli/pull/477.

### CODE REVIEW

**Review by MVrachev — CHANGES_REQUESTED** [review]
We have agreed that functionality, code optimizations, UI, docs and tests changes will be undressed with following prs.
The requirement for this pr is to @kairoaraujo, @lukpueh and me to agree on the user flow.

I will split my review by commands.
I have reviewed only sign and md update flows.

## Sign

Only two small comments:
1. Give information to the user whether and where the payload will be saved.
2. I think we can rename `-o` to `-s` and `----payload-out` to `--save-path` options.

## Metadata Update

1. When asking about threshold changes, the default value should be `n` or no.
2. Add a question asking for custom online key name
3. Give information to the user whether and where the payload will be saved.
4. I think we can rename `-o` to `-s` and `----payload-out` to `--save-path` options.

As a note for metadata update: we have decided to not use authorization, but the metadata update ceremony makes sure that you have at least one signature added.

Other improvements we have discussed will be documented in an issue.

I will review the ceremony flow in the next review.

File: repository_service_tuf/cli/admin2/sign.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    """Add one signature to root metadata."""
+    console.print("\n", Markdown("# Metadata Signing Tool"))
+
+    ###########################################################################
+    # Load roots
+    # TODO: load from API
+    root_md = Metadata[Root].from_bytes(root_in.read())
+
+    prev_root = None
+    if prev_root_in:
```
Comment (lukpueh) [inline repository_service_tuf/cli/admin2/sign.py:47]: TODO: raise if not prev_root_in and root_md version is `>1`,

**Comment by lukpueh** [comment]
> 3. Give information to the user on whether and where the payload will be saved.

@MVrachev, IIRC, we agreed that it's okay to merge this without API integration. What do think about, in the meanwhile, either
- warn, if the user does not provide the `-s` flag, or
- make the flag required?

Even thought the cli without API integration is experimental, we don't want to frustrate the user, by letting them go through the dialog without anything happening in the end.

**Comment by MVrachev** [comment]
> > 3. Give information to the user on whether and where the payload will be saved.
> 
> @MVrachev, IIRC, we agreed that it's okay to merge this without API integration. 

Yes, that's correct.

> What do think about, in the meanwhile, either
> 
> * warn, if the user does not provide the `-s` flag, or
> * make the flag required?

I am fine either way.
Maybe a warning will be better as we can also mention that the commands are experimental.
Then, when all of the features are added and we are replacing the current commands we would just remove one warning for each command.

> Even thought the cli without API integration is experimental, we don't want to frustrate the user, by letting them go through the dialog without anything happening in the end.

Agree.

File: repository_service_tuf/cli/admin2/sign.py (hunk 1 above)
Comment (lukpueh) [reply repository_service_tuf/cli/admin2/sign.py:50]: Fixed in e1ef9302c41e2739ad8c4c980a74b7d08b572bcb

**Review by MVrachev — APPROVED** [review]
All comments stopping this pr from merge were addressed.

We have a lot more to do, but this is a great start!

Awesome job @lukpueh!

PS: before merging we would have to make sure we are on top of the latest main changes and functional tests should pass.

**Comment by MVrachev** [comment]
As this is a big change: do you have to ask, add, or want to change anything before merging @kairoaraujo?

**Comment by kairoaraujo** [comment]
> As this is a big change: do you have to ask, add, or want to change anything before merging @kairoaraujo?

Not to add, it is isolated :)

**Pull request merged**